### PR TITLE
PIM-4940: Filter available UI locales from minimal translated progress

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php [hunk 1]
```diff
@@ -5,19 +5,52 @@
...
-                'choices' => Intl::getLocaleBundle()->getLocaleNames('en'),
-            )
-        );
+        $availableLocales = Intl::getLocaleBundle()->getLocaleNames('en');
+        $locales = [];
+
+        if (null == $this->translator) {
+            $locales = $availableLocales;
+        }
+        else {
```
Comment (skeleton) [inline src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php]: please write this syntax:
 `} else {`

File: src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php [hunk 2]
```diff
@@ -5,19 +5,52 @@
...
     {
-        $resolver->setDefaults(
-            array(
-                'choices' => Intl::getLocaleBundle()->getLocaleNames('en'),
-            )
-        );
+        $availableLocales = Intl::getLocaleBundle()->getLocaleNames('en');
+        $locales = [];
+
+        if (null == $this->translator) {
```
Comment (skeleton) [inline src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php]: in which case the service `translator` can be null ?
And if it's true, please make a strict condition with `===`

File: src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php [hunk 3]
```diff
@@ -5,19 +5,47 @@
...
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\Translator;
 
 class LocaleType extends AbstractType
 {
+    /** @var Translator */
+    protected $translator;
+
+    /** @var float */
+    protected $minTranslationProgress;
```
Comment (jmleroux) [inline src/Oro/Bundle/LocaleBundle/Form/Type/LocaleType.php:16]: `$minProgress` or `$minimumProgress` should be sufficient.

File: src/Oro/Bundle/LocaleBundle/Resources/config/services.yml [hunk 4]
```diff
@@ -65,6 +65,9 @@ services:
 
     oro_locale.form.type.locale:
         class: %oro_locale.form.type.locale.class%
+        arguments:
+            - @translator.default
+            - 0.8
```
Comment (jmleroux) [inline src/Oro/Bundle/LocaleBundle/Resources/config/services.yml:70]: should not be hardcoded

**Pull request closed without merging**